### PR TITLE
Upgrade Gradle to 6.9.2

### PR DESCRIPTION
**Pull request by yhtMinceraft1010X**
Part of #1661. Upgrading to version `6.9.2` is meant to be a stopgap measure while resolving errors from third-party plugins.

Propsed Commit Message:
```
Gradle has been stuck at 5.2.1 for quite some time. This version is not
compatible with a future migration to Java 17. However, upgrading to
Gradle 7.4 cannot be done yet due to a need to resolve errors from
third-party plugins.

Let's upgrade Gradle to 6.9.2 in the meantime to close the gap between
the version we are using and the latest available version.
```

### CODE REVIEW

**Comment by chan-j-d** [comment]
I've requested a second run of the deployment. It seems the previous build failed on this 
https://github.com/reposense/RepoSense/runs/5174576252?check_suite_focus=true#step:11:1465.
I've encountered it myself a few times and it's always somewhat random, perhaps we should raise a separate issue about it.

**Comment by yhtMinceraft1010X** [comment]
> I've requested a second run of the checks. It seems the previous build failed on this https://github.com/reposense/RepoSense/runs/5174576252?check_suite_focus=true#step:11:1465. I've encountered it myself a few times and it's always somewhat random, perhaps we should raise a separate issue about it.

Might be a concurrency issue similar to #1601. I think we should wait for #1625 to be merged.

File: build.gradle [hunk 1]
```diff
@@ -103,40 +103,44 @@ test {
...
 
-task installFrontend(type: com.liferay.gradle.plugins.node.tasks.ExecuteNpmTask) {
+tasks.register('installFrontend', com.liferay.gradle.plugins.node.tasks.ExecutePackageManagerTask) {
     workingDir 'frontend/'
     args = ['install']
 }
 
-task buildFrontend(dependsOn: ['installFrontend'], type: com.liferay.gradle.plugins.node.tasks.ExecuteNpmTask) {
+tasks.register('buildFrontend', com.liferay.gradle.plugins.node.tasks.ExecutePackageManagerTask) {
+    dependsOn 'installFrontend'
```
Comment (chan-j-d) [inline build.gradle:120]: Would it be possible to do an assignment like `def installFrontend = tasks.register(...)`? This way all further references to `installFrontend` like in line 120 can reference `installFrontend` without quotations directly. At least this seems possible when I tried it locally.

**Review by chan-j-d — APPROVED** [review]
LGTM! The current implementation seems to work well. Just some minor comments/clarifications.

Would like to confirm if rewriting the task definitions as `tasks.register(...)` is to support a future gradle 7 upgrade [described here](https://docs.gradle.org/7.4.1/userguide/more_about_tasks.html#sec:defining_tasks) since it seems the gradle 6.8.2 documentation uses quite a different syntax.
Additionally, would it be possible to assign the tasks to variables to avoid referencing them by string? More details are described in the specific comment.

File: build.gradle (hunk 1 above)
Comment (yhtMinceraft1010X) [reply build.gradle]: I managed to change some of the assignments

**Comment by yhtMinceraft1010X** [comment]
For `tasks.register(...)`, it's used for [task configuration avoidance](https://docs.gradle.org/6.9.2/userguide/task_configuration_avoidance.html). This prevents unnecessary tasks from being configured, which is best paired with #1631 - when only backend tests or backend checkstyle are run, configuration of frontend-related tasks like `installFrontend` is not carried out.

**Comment by gerhean** [comment]
May I ask about the significance of including the `Apache License` block of text? It's a little weird that it `Copyright 2015 the original author or authors.` but nowhere states who these author or authors are.

**Comment by dcshzj** [comment]
> May I ask about the significance of including the `Apache License` block of text? It's a little weird that it `Copyright 2015 the original author or authors.` but nowhere states who these author or authors are.

@gerhean I think the Apache License terms [requires that the copyright notice is retained](https://choosealicense.com/licenses/apache-2.0/). I think for the "the original author or authors", we should leave it as it is, in case of licensing issues that may occur.

**Comment by gerhean** [comment]
Just want to check that this has been tested to work on WIndows?

**Comment by chan-j-d** [comment]
> Just want to confirm that this has been tested to work on Windows? I assume it works on MacOs and Linux since the CI passed.

@gerhean Yep, I pulled the branch, `clean build`, `frontendTest` and ran it against a few repositories when I tested it.

**Review by gerhean — APPROVED** [review]
LGTM

**Review by fzdy1914 — APPROVED** [review]

**Pull request merged**